<a href="https://colab.research.google.com/github/michaelherediaperez/mecanica_de_solidos_un/blob/main/codigos/cap_02/02_08_01_ejemplo_01.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Ejemplo sección 2.8.1

|Quién | Fecha | Qué hizo |
| ---  | ---   | ---      |
|Michael Heredia Pérez, <mherediap@unal.edu.co> | Julio 14, 2022     | Conversión de código de MAXIMA a Python|
|Diego Andrés Alvarez, <daalvarez@unal.edu.co>  | Septiembre 7, 2022 | Mejorando comentarios                  |


Considere un puto de un sólido bidimensional en el cual los esfuerzos son $\sigma_x=$ 3 Pa, $\sigma_y=$ 2 Pa y $\tau_{xy}=$ -4 Pa. Se pide:

* Plantear la matriz de tensiones $\underline{\underline{\boldsymbol{\sigma}}}$ correspondiente.
* Calcular el polinomio característico asociado a $\underline{\underline{\boldsymbol\sigma}}$.
* Calcular la dirección y magnitud de los esfuerzos principales.

In [1]:
import numpy as np  # Librería para cálculo algebráico

In [2]:
# Para mejorcomprensión, defino los componentes de la matriz sigma.
sx  =  3 
sy  =  2
txy = -4

# Defino la matriz sigma
sigma = np.array([[ sx,  txy],
                  [txy,  sy]])

$\underline{\underline{\sigma}}=$

In [3]:
sigma

array([[ 3, -4],
       [-4,  2]])

Se calcula el polinomio característico:

In [4]:
np.poly(sigma)

array([  1.,  -5., -10.])

Lo cual quiere decir que el polinomio característico es: $\sigma_n^2 - 5\sigma_n - 10 = 0$.

Se calculan los valores propios de $\underline{\underline{\boldsymbol\sigma}}$ (los esfuerzos principales) con ```numpy.linalg.eigvals()```:

In [5]:
np.linalg.eigvals(sigma)

array([ 6.53112887, -1.53112887])

Es decir $(\sigma_1)_{xy} = 6.5311$ Pa y $(\sigma_2)_{xy} = -1.5311$ Pa. 

Observe que los valores propios también se obtienen calculando las raíces del polinomio característico:

In [6]:
np.roots(np.poly(sigma))

array([ 6.53112887, -1.53112887])

Para el cálculo de las direcciones principales, se requiere calcular los vectores propios de la matriz $\underline{\underline{\boldsymbol\sigma}}$, lo cual se realiza con el comando ```eig```:

In [7]:
valp, vecp = np.linalg.eig(sigma)

Los valores propios representan la magnitud de los esfuerzos principales,

$[(\sigma_1)_{xy}, [(\sigma_2)_{xy}] = $  

In [8]:
valp

array([ 6.53112887, -1.53112887])

Los vectores propios representan la dirección de los esfuerzos principales.

In [9]:
vecp

array([[ 0.74967818,  0.66180256],
       [-0.66180256,  0.74967818]])

$\hat{\boldsymbol{n}}_1^T = $

In [10]:
vecp[0]

array([0.74967818, 0.66180256])

$\hat{\boldsymbol{n}}_2^T = $

In [11]:
vecp[1]

array([-0.66180256,  0.74967818])